coc/events.py

```diff
@@ -363,6 +365,115 @@ async def wrapped(cached_war: ClanWar, war: ClanWar, callback):
                 await callback(war)
         return _ValidateEvent.shortcut_register(wrapped, tags, custom_class, retry_interval, WarEvents.event_type)
 
+@_ValidateEvent
+class RaidEvents:
+    """Class that defines all valid raid events"""
+    event_type = "raid"
+
+    @classmethod
+    def new_member(cls, tags=None, custom_class=None, retry_interval=None):
+        """Event for when a new member has joined the raid weekend."""
+
+        async def wrapped(cached_raid: RaidLogEntry, raid: RaidLogEntry, callback):
+            if cached_raid.members:
+                new_members = (member for member in raid.members if member not in set(cached_raid.members))
```

Comment:
first I don't think that works. Due to the definition of the `__eq__` of raid member, this would also include all members with a new attack etc.
And even if it would work, why do you not do simply set(raid.members)-set(cached_raid.members)?

Reply:
I think the proper way to do that would be `(member for member in raid.members if member.tag not in [m.tag for m in cached_raid.members])`